src/zenml/zen_stores/rest_zen_store.py

```diff
@@ -252,9 +266,20 @@ def copy_configuration(
             A new store configuration object that reflects the new configuration
             path.
         """
+        from zenml.utils.docker_utils import (
+            replace_localhost_with_internal_hostname,
+        )
+
         assert isinstance(config, RestZenStoreConfiguration)
         config = config.copy(deep=True)
 
+        # If the URL is localhost, then we make the assumption that the
+        # copied configuration will be used in a docker container running
+        # on the same machine (other scenarios are not technically possible)
```

Comment:
I don't know where exactly this method is called, but why are other scenarios not possible?

Reply:
The method is only called when docker images are built for orchestrators and operators that rely on containers. The technical impossibility comes from the range of orchestrators and operators that are available at the moment. They either use Docker locally (in which case the localhost mapping works as it should) or they use a remote cloud (in which case, the setup is not going to work either way, so it's irrelevant what we do here).